src/main/java/org/springframework/data/jpa/repository/query/QueryUtils.java

```diff
@@ -202,11 +203,12 @@ public static String getExistsQueryString(String entityName, String countQueryPl
 	/**
 	 * Returns the query string for the given class name.
 	 *
-	 * @param template
-	 * @param entityName
-	 * @return
+	 * @param template must not be {@literal null}.
+	 * @param entityName must not be {@literal null}.
+	 * @return the template with placeholders replaced by the {@literal entityName}. Guaranteed to be not {@literal null}.
 	 */
-	public static String getQueryString(String template, String entityName) {
+	@NonNull
```

Comment:
Why do we need explicit `@NonNull` annotations here? Isn't that the default anyway?

Reply:
Wasn't aware that `@NonNull` is the default, I thought that was controlled by a package level annotation. Removing `@NonNull`s again.